components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java

```diff
@@ -325,10 +325,13 @@ public boolean doPostAuthenticate(String userName, boolean authenticated,
                         String emailTemplate = null;
                         int tenantId = userStoreManager.getTenantId();
                         String firstName = null;
+                        String userStoreDomain = userStoreManager.getRealmConfiguration()
+                                .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+                        String domainSpecificUserName = UserCoreUtil.addDomainToName(userName, userStoreDomain);
+
                         try {
-                            firstName =
-                                    Utils.getClaimFromUserStoreManager(userName, tenantId,
-                                            "http://wso2.org/claims/givenname");
+                            firstName = Utils.getClaimFromUserStoreManager(domainSpecificUserName, tenantId,
+                                    "http://wso2.org/claims/givenname");
```

Comment:
Actually the givenname constant is defined in user.core in kernel as a constant [1]. However it is OK to introduce a new claim in identity.mgt. 

[1] https://github.com/wso2/carbon-kernel/blob/4.4.x/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UserCoreConstants.java#L204